datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1430,4 +1437,55 @@ public FlowIdentifier updateSalt(SdxCluster sdxCluster) {
             return sdxReactorFlowManager.triggerSaltUpdate(sdxCluster);
         }
     }
+
+    public FlowIdentifier horizontalScaleDatalake(String name, DatalakeHorizontalScaleRequest scaleRequest) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+//        if (!entitlementService.isDatalakeHorizontalScaleEnabled(accountId)) {
+//            throw new BadRequestException("Datlake horizontal scale is not enabled");
+//        }
+
+        LOGGER.info("Horizontal scaling Data Lake with name {}", name);
+        final SdxCluster sdxCluster = sdxClusterRepository.findByAccountIdAndClusterNameAndDeletedIsNullAndDetachedIsFalse(accountId, name)
+                .orElseThrow(() -> notFound("SDX cluster", name).get());
+        return sdxReactorFlowManager.triggerHorizontalScaleDataLake(sdxCluster, scaleRequest);
+    }
+
+    public void validateHorizontalScaleRequest(SdxCluster sdxCluster, DatalakeHorizontalScaleRequest request) {
+        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
+            throw new BadRequestException(String.format("scaling not supported on: %s.", sdxCluster.getClusterShape().name()));
+        }
+        var hostGroup = request.getGroup();
+        if (!DatalakeInstanceGroupName.valueOf(hostGroup.toUpperCase(Locale.ROOT)).isScalable()) {
+            throw new BadRequestException(String.format("%s host group cannot scale", hostGroup));
+        }
+        DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(sdxCluster.getEnvName());
+        if (!EnvironmentStatus.upscalable().contains(environmentResponse.getEnvironmentStatus())) {
+            throw new BadRequestException(String.format("The cluster cannot be scaled in the %s status.", environmentResponse.getEnvironmentStatus().name()));
+        }
+        StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
+        if (isNodeRequestInvalid(stackResponse, request)) {
+            throw new BadRequestException("The requested nodeCount does not reach the minimum condition of horizontal scaling.");
+        }
+    }
+
+    private boolean isNodeRequestInvalid(StackV4Response stack, DatalakeHorizontalScaleRequest request) {
+        if (request.getDesiredCount() < 0) {
+            LOGGER.warn("Negative nodeCount was requested.");
+            return true;
+        }
+        var hostgroupMap = stack.getInstanceGroups()
+                .stream()
+                .filter(group -> group.getName().equalsIgnoreCase(request.getGroup()))
+                .collect(toMap(InstanceGroupV4Response::getName, InstanceGroupV4Response::getNodeCount));
```

Comment:
nitpick - no need to collect into a container if you want to check something just exists